examples/morpheusvm/actions/transfer_test.go

```diff
@@ -0,0 +1,137 @@
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: 1,
+			},
+			State:       ts.NewView(make(state.Keys), map[string][]byte{}),
+			ExpectedErr: tstate.ErrInvalidKeyOrPermission,
+		},
+		{
+			Name:  "NotEnoughBalance",
+			Actor: codec.EmptyAddress,
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: 1,
+			},
+			State: func() state.Mutable {
+				keys := make(state.Keys)
+				keys.Add(string(emptyBalanceKey), state.Read)
+				tsv := ts.NewView(keys, map[string][]byte{})
+				return tsv
+			}(),
+			ExpectedErr: storage.ErrInvalidBalance,
+		},
+		{
+			Name:  "SelfTransfer",
+			Actor: codec.EmptyAddress,
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: 1,
+			},
+			State: func() state.Mutable {
+				keys := make(state.Keys)
+				store := chaintest.NewInMemoryStore()
+				req.NoError(storage.SetBalance(context.Background(), store, codec.EmptyAddress, 1))
+				keys.Add(string(emptyBalanceKey), state.All)
+				return ts.NewView(keys, store.Storage)
+			}(),
+			Assertion: func(ctx context.Context, t *testing.T, store state.Mutable) {
+				require := require.New(t)
+				balance, err := storage.GetBalance(ctx, store, codec.EmptyAddress)
+				require.NoError(err)
+				require.Equal(balance, uint64(1))
+			},
+		},
+		{
+			Name:  "OverflowBalance",
+			Actor: codec.EmptyAddress,
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: math.MaxUint64,
+			},
+			State: func() state.Mutable {
+				keys := make(state.Keys)
+				store := chaintest.NewInMemoryStore()
+				req.NoError(storage.SetBalance(context.Background(), store, codec.EmptyAddress, 1))
+				keys.Add(string(emptyBalanceKey), state.All)
+				return ts.NewView(keys, store.Storage)
+			}(),
+			ExpectedErr: storage.ErrInvalidBalance,
+		},
+		{
+			Name:  "SimpleTransfer",
+			Actor: codec.EmptyAddress,
+			Action: &Transfer{
+				To:    oneAddr,
+				Value: 1,
+			},
+			State: func() state.Mutable {
+				keys := make(state.Keys)
+				store := chaintest.NewInMemoryStore()
+				req.NoError(storage.SetBalance(context.Background(), store, codec.EmptyAddress, 1))
+				keys.Add(string(emptyBalanceKey), state.All)
+				keys.Add(string(storage.BalanceKey(oneAddr)), state.All)
+				return ts.NewView(keys, store.Storage)
+			}(),
+			Assertion: func(ctx context.Context, t *testing.T, store state.Mutable) {
+				require := require.New(t)
+				receiverBalance, err := storage.GetBalance(ctx, store, oneAddr)
+				require.NoError(err)
+				senderBalance, err := storage.GetBalance(ctx, store, codec.EmptyAddress)
+				require.NoError(err)
+				require.Equal(receiverBalance, uint64(1))
+				require.Equal(senderBalance, uint64(0))
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		tt.Run(context.Background(), t)
+	}
+}
+
+func createAddressWithByte(b byte) (codec.Address, error) {
+	addrSlice := make([]byte, codec.AddressLen)
+	for i := range addrSlice {
+		addrSlice[i] = b
+	}
+	return codec.ToAddress(addrSlice)
+}
```

Comment:
Optional for this PR - we should move this utility to a separate shared test package since other VMs will need this as well

Reply:
Opened https://github.com/ava-labs/hypersdk/issues/1186